test_integration/lib/test/rest/rest_request_token_test.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:ably_flutter/ably_flutter.dart';
+import 'package:ably_flutter_integration_test/app_provisioning.dart';
+import 'package:ably_flutter_integration_test/factory/reporter.dart';
+
+Future<Map<String, dynamic>> testRestRequestToken({
+  required Reporter reporter,
+  Map<String, dynamic>? payload,
+}) async {
+  reporter.reportLog('init start');
+  final appKey = await AppProvisioning().provisionApp();
+
+  //init ably for token
+  final clientOptionsForToken = ClientOptions(
+    key: appKey,
+    environment: 'sandbox',
+    logLevel: LogLevel.verbose,
+  );
+
+  final rest = Rest(
+    options: clientOptionsForToken,
+  );
+
+  final token = await rest.auth?.requestToken();
+
+  if (token == null) {
```

Comment:
I know we probably don't need to test these auth methods extensively since they're already tested by the underlying libraries but I think we ought to do a bit more than just checking that they return something. For example I think it would be better to test that you can successfully initialise a new realtime client using a token obtained from `requestToken`

Reply:
Removed some unnecessary tests and improved some 4768cf3966977460616432dc0c8da88adf997b5d and 6ac176204355356fe63d82b857e253255117919b